src/v/storage/disk_log_impl.cc

```diff
@@ -79,6 +79,12 @@ disk_log_impl::~disk_log_impl() {
     vassert(_closed, "log segment must be closed before deleting:{}", *this);
 }
 
+ss::future<> disk_log_impl::recompute_max_segment_size() {
+    _max_segment_size = internal::jitter_segment_size(
+      std::min(max_segment_size(), segment_size_hard_limit));
+    return ss::make_ready_future<>();
```

Comment:
Is it important that this is a future? If so, could you leave a comment explaining why?

Reply:
A guess is that If invoked in a loop a future yields.

Reply:
i think it doesn't have to be a future in the context of how the method is used

Reply:
Thanks!

Loops aside, it does look like `jitter_segment_size()` isn't such a cheap method -- it generates some RNG and does some division. Not sure how much this'd show on a profile, but still worth noting; maybe we can cache the jitter per topic partition or somesuch instead of recomputing every roll.

Reply:
That's a fair point I think. I did it as a separate commit just in case anyone else has a strong opinion against caching the jitter.